app/ScheduledMessage.php

```diff
@@ -61,4 +61,16 @@ public function getHtmlAttribute()
 
         return (new LinkFinder)->process($this->attributes[$key]);
     }
+
+    public function sanitizeBody()
+    {
+        $this->body_en = gsmSanitize($this->body_en);
+        $this->body_es = gsmSanitize($this->body_es);
+        $this->save();
+    }
+}
+
+function gsmSanitize($in)
+{
+    return str_replace(['—', '-', '–'], '-', $in);
```

Comment:
This library might be worth a try instead of rolling our own sanitizer: https://github.com/BenMorel/GsmCharsetConverter looks like it supports Spanish characters too